src/containerapp/azext_containerapp/_client_factory.py

```diff
@@ -0,0 +1,91 @@
+            if 'message' in content:
+                detail = content['message']
+            elif 'Message' in content:
+                detail = content['Message']
+
+            if creating_plan:
+                if 'Requested features are not supported in region' in detail:
+                    detail = ("Plan with linux worker is not supported in current region. For " +
+                              "supported regions, please refer to https://docs.microsoft.com/"
+                              "azure/app-service-web/app-service-linux-intro")
+                elif 'Not enough available reserved instance servers to satisfy' in detail:
+                    detail = ("Plan with Linux worker can only be created in a group " +
+                              "which has never contained a Windows worker, and vice versa. " +
+                              "Please use a new resource group. Original error:" + detail)
+            ex = CLIError(detail)
+        except Exception:  # pylint: disable=broad-except
+            pass
+        if no_throw:
+            return ex
+        raise ex
+    return _polish_bad_errors
+
+
+def handle_raw_exception(e):
+    import json
+
+    stringErr = str(e)
+
+    if "{" in stringErr and "}" in stringErr:
+        jsonError = stringErr[stringErr.index("{"):stringErr.rindex("}") + 1]
+        jsonError = json.loads(jsonError)
+
+        if 'error' in jsonError:
+            jsonError = jsonError['error']
+
+            if 'code' in jsonError and 'message' in jsonError:
+                code = jsonError['code']
+                message = jsonError['message']
+                raise CLIError('({}) {}'.format(code, message))
+        elif "Message" in jsonError:
+            message = jsonError["Message"]
+            raise CLIError(message)
+        elif "message" in jsonError:
+            message = jsonError["message"]
+            raise CLIError(message)
+    raise e
+
+
+def providers_client_factory(cli_ctx, subscription_id=None):
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, subscription_id=subscription_id).providers
+
+
+def cf_resource_groups(cli_ctx, subscription_id=None):
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES,
+                                   subscription_id=subscription_id).resource_groups
+
+def log_analytics_client_factory(cli_ctx):
+    from azure.mgmt.loganalytics import LogAnalyticsManagementClient
+
+    return get_mgmt_service_client(cli_ctx, LogAnalyticsManagementClient).workspaces
+
+def log_analytics_shared_key_client_factory(cli_ctx):
+    from azure.mgmt.loganalytics import LogAnalyticsManagementClient
+
+    return get_mgmt_service_client(cli_ctx, LogAnalyticsManagementClient).shared_keys
+
+def cf_containerapp(cli_ctx, *_):
+
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    # TODO: Replace CONTOSO with the appropriate label and uncomment
```

Comment:
NIT - is this TODO needed? If so I don't understand the need for this.

Reply:
Removed, good catch